src/main/java/io/nats/client/impl/SocketDataPortWithWriteTimeout.java

```diff
@@ -36,12 +36,16 @@ class WriteWatcherTask extends TimerTask {
         public void run() {
             //  if now is after when it was supposed to be done by
             if (System.nanoTime() > writeMustBeDoneBy) {
+                writeWatcherTimer.cancel(); // we don't need to repeat this
+                connection.executeCallback((c, el) -> el.socketWriteTimeout(c));
                 try {
-                    connection.closeSocket(true);
+                    out.close();
                 }
-                catch (InterruptedException e) {
-                    // do nothing
+                catch (IOException ignore) {}
```

Comment:
try closing the output. Most likely going to fail since it was blocked, hence the catch, but also could have actually be freed up in the middle, so close it anyway, it's state is bad.